### ISSUE TITLE
switch to CSS-modules via webpack

### ISSUE THREAD

**Issue by jokeyrhyme** [unicorn-standard/starter-kit#11]
Would you accept a PR that uses [css-modules](https://github.com/css-modules/css-modules) via the [css-loader](https://github.com/webpack/css-loader#css-modules) webpack plugin that you're already using?

This may or may not be a transition from [pacomo](http://unicornstandard.com/packages/pacomo.html).

This approach would still have 1 CSS file next to the JSX file for a component, but relies on the build step to give each module its own private CSS class scope. This means you don't need a naming convention.

**Comment by jamesknelson** [comment]
@jokeyrhyme I do like CSS modules, but my react-pacomo package provides a few nicetes which CSS modules doesn't, namely:
- It allows you to pass an object of booleans to `className` (using the same format as the venerable classnames package)
- It automatically adds a class to the component's root element

If you there is a way to provide this kind of "batteries included" solution using CSS modules, I'd be interested - but no guarantees. I'd like to see an example first.

**Comment by jokeyrhyme** [comment]
Yeah, that was definitely something I missed when I switched my fork over to CSS modules. I've had to pull in the https://www.npmjs.com/package/classnames module and use it explicitly in a few places.

I might look into how you've hooked this all up and see if the react-pacomo integration you have can be paired back just those 2 convenient features.

**Comment by jamesknelson** [comment]
It certainly wouldn't hurt to split out react-pacomo into a few separate parts, as @pcwinters suggested over at the react-pacomo project. That said, I've been way too busy to look at this lately. Hopefully will get some free time at some point.

**Comment by jokeyrhyme** [comment]
I found these related modules:
- https://www.npmjs.com/package/react-classnames
- https://www.npmjs.com/package/react-classname-module

With some PRs to add tests, they might cover some of what we need here.
